kotlin/group-layers/src/main/java/com/esri/arcgisruntime/sample/grouplayers/LayerListAdapter.kt

```diff
@@ -0,0 +1,172 @@
+    val layer = dataSet[position]
+    return if (layer is GroupLayer && layer.visibilityMode == GroupVisibilityMode.EXCLUSIVE) {
+      TYPE_EXCLUSIVE
+    } else TYPE_DEFAULT
+  }
+
+  override fun getItemCount(): Int = dataSet.size
+
+  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      TYPE_EXCLUSIVE -> (holder as ExclusiveLayerViewHolder).let { exclusiveLayerViewHolder ->
+        val layer = dataSet[position]
+        exclusiveLayerViewHolder.layer = layer
+        exclusiveLayerViewHolder.textView.text = layer.name
+        exclusiveLayerViewHolder.onLayerChecked = onLayerCheckedChanged
+        exclusiveLayerViewHolder.sublayers.apply {
+          clear()
+          addAll((layer as GroupLayer).layers)
+        }
+        exclusiveLayerViewHolder.populate()
+      }
+      else -> (holder as DefaultLayerViewHolder).let { defaultLayerViewHolder ->
+        val layer = dataSet[position]
+        defaultLayerViewHolder.layer = layer
+        defaultLayerViewHolder.textView.text = layer.name
+        defaultLayerViewHolder.onLayerChecked = onLayerCheckedChanged
+        defaultLayerViewHolder.sublayers.apply {
+          clear()
+          addAll((layer as GroupLayer).layers)
+        }
+        defaultLayerViewHolder.populate()
+      }
+    }
+  }
+
+
+  abstract class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+    abstract val textView: TextView
+  }
+
+  class DefaultLayerViewHolder(itemView: View) : ViewHolder(itemView) {
+    override val textView: TextView = itemView.findViewById(
+      R.id.checkbox_grouplayer_name
+    )
+    private val sublayerLayout: LinearLayout = itemView.findViewById(
+      R.id.sublayer_layout
+    )
+    val sublayers = mutableListOf<Layer>()
+    var layer: Layer? = null
+
+    var onLayerChecked: ((Layer, Boolean) -> Unit)? = null
+
+    fun populate() {
+      val checkBox: CheckBox = itemView.findViewById(R.id.checkbox)
+      layer?.let { layer ->
+        checkBox.setOnCheckedChangeListener { _, isChecked ->
+          onLayerChecked?.invoke(layer, isChecked)
+        }
+        checkBox.isChecked = layer.isVisible
+      }
+      sublayers.forEach { sublayer ->
+        CheckBox(itemView.context).apply {
+          id = View.generateViewId()
+          text = sublayer.name
+          sublayerLayout.addView(this)
+          setOnCheckedChangeListener { _, isChecked -> onLayerChecked?.invoke(sublayer, isChecked) }
+          isChecked = sublayer.isVisible
+        }
+      }
+    }
+  }
+
+  class ExclusiveLayerViewHolder(itemView: View) : ViewHolder(itemView) {
+    override val textView: TextView = itemView.findViewById(
+      R.id.radio_group_layer_name
+    )
+    private val radioGroup: RadioGroup = itemView.findViewById(
+      R.id.radioGroup
+    )
+    val sublayers = mutableListOf<Layer>()
+    var layer: Layer? = null
+
+    var onLayerChecked: ((Layer, Boolean) -> Unit)? = null
+
+    fun populate() {
```

Comment:
Some KDoc would be nice here and for the other populate below

Reply:
sure thing!